### DISCUSSION TITLE
YAML schema to make editing files easier

### DISCUSSION THREAD

**Discussion by cverdes** [lowdefy/lowdefy#1038] [Questions and Help] [answered]
Dear community, 

Does it exist a YAML a schema config file for Lowdefy syntax in order to use with the _YAML Language Support by Red Hat extension_ with Visual Studio Code?

As i can understand (maybe I'm wrong) in the [extension web page](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) says that different schema syntax can be used created under the [JSON Schema specifications](https://json-schema.org/).

If a syntax scheme is defined, the extension provides Document Outlining, Auto completion, YAML validation and more.

[This article](https://developers.redhat.com/blog/2020/11/25/how-to-configure-yaml-schema-to-make-editing-files-easier#) explain it with more details how to configure YAML schema to make editing files easier.

Has anyone used this extension with a Lowdefy syntax scheme?

Thanks in advance.

**Comment by cverdes** [comment]
Sorry, reading more about YAML Language Support by Red Hat extension, I've discovered the [YAML Schema ASDF Standard](https://asdf-standard.readthedocs.io/en/latest/schemas/yaml_schema.html), and realised that it's seems to be the same that the blocks JSON Schema Definitions included in the Lowdefy documentation.

Have anyone used the blocks JSON Schema Definitions included in the Lowdefy documentation for syntax validation and autocomplete in VSCode? How the settings.json file have to be configured?

Thanks in advance.

**Comment by SamTolmay** [comment, marked as answer]
Hi @cverdes 

I would be really great if we could get this working. There are some difficulties relating to the `_ref` operator for example, but we have not been able to look at this in detail yet. You can have a look at this discussion regarding this (#677), and maybe add some thoughts there.

**Reply by cverdes** [reply]
Thank for your answer @SamTolmay.  
I read the discussion regarding the (#677), and I agree with @Gervwyk's idea that a Lowdefy language server seems like a better way to go. Although more complex, with much better potential.
I will study the VSCode documentation about how to implement a Language Server and I'll try to do a small conceptual prototype to show you.
I hope I can move forward and achieve some concrete result. I'll share my research in the (#677) discussion.
Best Regards
